### PR TITLE
feat: better support for zig versions

### PR DESCRIPTION
**Pull request by weskoerber**
Follow-up from https://github.com/marlersoft/zigwin32/pull/31.

### CODE REVIEW

File: src/zigwin32.build.zig [hunk 1]
```diff
@@ -1,16 +1,27 @@
 const builtin = @import("builtin");
 const std = @import("std");
 
-const is_zig_0_11 = std.mem.eql(u8, builtin.zig_version_string, "0.11.0");
+const zig_version = builtin.zig_version;
```
Comment (marler8997) [inline src/zigwin32.build.zig:4]: If we're gonna use a switch below, we can remove this line/variable

File: src/zigwin32.build.zig [hunk 2]
```diff
@@ -1,16 +1,27 @@
...
+            _ = b.addModule("zigwin32", .{
+                .source_file = .{ .path = "win32.zig" },
+            });
+        },
+        12 => {
+            _ = b.addModule("zigwin32", .{
+                .root_source_file = .{ .path = "win32.zig" },
+            });
+        },
+        13 => {
```
Comment (marler8997) [inline src/zigwin32.build.zig:18]: Let's make this the else, there's a non zero chance the zigwin32 code today will work with future zig versions, it's just bindings and a few lines of build.zig.

File: src/zigwin32.build.zig (hunk 1 above)
Comment (weskoerber) [reply src/zigwin32.build.zig]: Sounds good!

File: src/zigwin32.build.zig (hunk 2 above)
Comment (weskoerber) [reply src/zigwin32.build.zig]: Gotcha, sounds good!

**Comment by Tired-Fox** [comment]
v0.12.0 also requires `b.path` to be used for `root_source_file`

**Comment by weskoerber** [comment]
> v0.12.0 also requires `b.path` to be used for `root_source_file`

It's technically not required but it is deprecated (see 0.12.0 release notes [here](https://ziglang.org/download/0.12.0/release-notes.html#introduce-bpath-deprecate-LazyPathrelative)). This change was done in https://github.com/ziglang/zig/pull/19597.

With that said, the switch could definitely be simplified as 0.12.0 supports `b.path()`. We could just check if the minor version is 11 and use the `path` field, otherwise use `b.path()`.

**Comment by marler8997** [comment]
I incorporated the change in this commit: https://github.com/marlersoft/zigwin32gen/commit/db32e7c0cb3ed5645b4a9a2bf5814e88def0a89b

However I dropped support for 0.11.0, people on that zig version can still use old versions of zigwin32.

**Pull request closed without merging**